### PR TITLE
Streams: settle reader.[[closedPromise]] before performing close/error steps of read requests

### PR DESCRIPTION
**Pull request by MattiasBuelens**
Accompanies whatwg/streams#1102

### CODE REVIEW

**Review by domenic — APPROVED** [review]
LGTM but we shouldn't merge the general.any.js change until the spec change is merged.

File: streams/readable-streams/async-iterator.any.js
Comment (ricea) [inline streams/readable-streams/async-iterator.any.js:640]: Could you check that reader.closed has changed to a different promise?

Comment (MattiasBuelens) [reply streams/readable-streams/async-iterator.any.js:640]: No, I can't check it here. This reader is internal to the async iterator, and the iterator implementation does not use `[[closedPromise]]` in any way.

That said, it doesn't look like this is covered by any of our existing tests for "regular" readers. If you want, I can copy [these tests](https://github.com/web-platform-tests/wpt/blob/9378cebe0d791f016d1b0ecca105e10941d7a34f/streams/readable-streams/default-reader.any.js#L117-L155) and check that `reader.closed` changes after calling `reader.releaseLock()`?

Comment (ricea) [reply streams/readable-streams/async-iterator.any.js:640]: > No, I can't check it here. This reader is internal to the async iterator, and the iterator implementation does not use `[[closedPromise]]` in any way.

Oops, yes, sorry.

> That said, it doesn't look like this is covered by any of our existing tests for "regular" readers. If you want, I can copy these tests and check that `reader.closed` changes after calling `reader.releaseLock()`?

That would be great, thanks.

**Comment by domenic** [comment]
Oh, shoot, I forgot that @ricea had substantive comments on the last round. I should let him have another look at this to give his LGTM before merging.

**Review by ricea — APPROVED** [review]
lgtm from me too.

**Pull request merged**
